File: cpp/GudangGadget.cpp
#include <iostream>
#include <iomanip>
#include <string>
using namespace std;

class GudangGadget {
private:
    string idGudang;
    string namaGadget;
    double harga;
    int jumlahStok;
    string kategoriTeknologi;
    int garansiBulan;
    string kondisi;

public:
    // Konstruktor
    GudangGadget(string id, string nama, double hrg, int stok,string kategori, int garansi, string kond) {
        this -> idGudang = id;
        this -> namaGadget = nama;
        this ->harga = hrg;
        this ->jumlahStok = stok;
        this ->kategoriTeknologi = kategori;
        this ->garansiBulan = garansi;
        this ->kondisi = kond;
    }

    // Getter
    string getIdGudang() const{

     return idGudang;
     
    }
    string getNamaGadget()const{

       return namaGadget;
    }
    double getHarga(){

       return harga; 
    } 
    int getJumlahStok() {

     return jumlahStok;
    }
    string getKategoriTeknologi(){

        return kategoriTeknologi; 
    } 
    int getGaransiBulan() {

        return garansiBulan; 
    }
    string getKondisi() {

        return kondisi; 
    }

    // Setter
    void setNamaGadget(string nama) { 
        this -> namaGadget = nama;
     }
    void setHarga(double hrg) {
        this-> harga = hrg; 
    }
    void setJumlahStok(int stok) {
        this-> jumlahStok = stok;
     }
    void setKategoriTeknologi(string kategori) { 
        this -> kategoriTeknologi = kategori; 
    }
    void setGaransiBulan(int garansi) {
        this -> garansiBulan = garansi;
     }
    void setKondisi(string kond) { 
        this -> kondisi = kond;
     }

    // Method tampilkan info
    void tampilkanInfo() const {
        cout << "*******************************" << endl;
        cout << "* ID Gudang     : " << idGudang << endl;
        cout << "* Nama Gadget   : " << namaGadget << endl;
        cout << "* Harga         : Rp " << fixed << setprecision(2) << harga << endl;
        cout << "* Jumlah Stok   : " << jumlahStok << endl;
        cout << "* Kategori      : " << kategoriTeknologi << endl;
        cout << "* Garansi       : " << garansiBulan << " bulan" << endl;
        cout << "* Kondisi       : " << kondisi << endl;
        cout << "*******************************" << endl;
    }
};

File: java/GudangGadget.java
public class GudangGadget {
    private String idGudang;
    private String namaGadget;
    private double harga;
    private int jumlahStok;
    private String kategoriTeknologi;
    private int garansiBulan;
    private String kondisi;

    // Konstruktor
    public GudangGadget(String id, String nama, double hrg, int stok, String kategori, int garansi, String kond) {
        this.idGudang = id;
        this.namaGadget = nama;
        this.harga = hrg;
        this.jumlahStok = stok;
        this.kategoriTeknologi = kategori;
        this.garansiBulan = garansi;
        this.kondisi = kond;
    }

    // Getter
    public String getIdGudang() { return idGudang; }
    public String getNamaGadget() { return namaGadget; }
    public double getHarga() { return harga; }
    public int getJumlahStok() { return jumlahStok; }
    public String getKategoriTeknologi() { return kategoriTeknologi; }
    public int getGaransiBulan() { return garansiBulan; }
    public String getKondisi() { return kondisi; }

    // Setter
    public void setNamaGadget(String nama) { this.namaGadget = nama; }
    public void setHarga(double hrg) { this.harga = hrg; }
    public void setJumlahStok(int stok) { this.jumlahStok = stok; }
    public void setKategoriTeknologi(String kategori) { this.kategoriTeknologi = kategori; }
    public void setGaransiBulan(int garansi) { this.garansiBulan = garansi; }
    public void setKondisi(String kond) { this.kondisi = kond; }

    // Method tampilkan info
    public void tampilkanInfo() {
        System.out.println("*******************************");
        System.out.println("* ID Gudang     : " + idGudang);
        System.out.println("* Nama Gadget   : " + namaGadget);
        System.out.printf("* Harga         : Rp %.2f%n", harga);
        System.out.println("* Jumlah Stok   : " + jumlahStok);
        System.out.println("* Kategori      : " + kategoriTeknologi);
        System.out.println("* Garansi       : " + garansiBulan + " bulan");
        System.out.println("* Kondisi       : " + kondisi);
        System.out.println("*******************************");
    }
}
